### DISCUSSION TITLE
amp import error

### DISCUSSION THREAD

**Discussion by shengzhang90** [Project-MONAI/tutorials#1875] [General] [answered]
Hi, all

When I run the maisi_train_vae_tutorial.ipynb file, I encounter the following issue.

ImportError: cannot import name 'GradScaler' from 'torch.amp' (/home/shengzhang/miniconda3/envs/maisi/lib/python3.10/site-packages/torch/amp/__init__.py)

I have executed the following command "conda install -c conda-forge nvidia-apex", but it does not work.

Thanks a lot.

**Comment by KumoLiu** [comment]
Hi @shengzhang90, may I know your torch version.
torch.cuda.amp has been marked as deprecated, and the new torch.amp API is recommended starting with PyTorch 2.0, torch.amp is introduced since torch 1.10.
Thanks.

**Reply by shengzhang90** [reply]
Hi, @KumoLiu Thanks a lot for your responsive answer.

The following is my virtual environment:

MONAI version: 1.4.0
Numpy version: 1.25.2
Pytorch version: 2.1.1+cu121
MONAI flags: HAS_EXT = False, USE_COMPILED = False, USE_META_DICT = False
MONAI rev id: 46a5272196a6c2590ca2589029eed8e4d56ff008
MONAI __file__: /home/<username>/miniconda3/envs/maisi/lib/python3.10/site-packages/monai/__init__.py

Optional dependencies:
Pytorch Ignite version: 0.4.11
ITK version: 5.4.0
Nibabel version: 5.3.2
scikit-image version: 0.24.0
scipy version: 1.14.1
Pillow version: 10.4.0
Tensorboard version: 2.18.0
gdown version: 5.2.0
TorchVision version: 0.16.1+cu121
tqdm version: 4.67.0
lmdb version: 1.5.1
psutil version: 6.1.0
pandas version: 2.2.3
einops version: 0.8.0
transformers version: 4.40.2
mlflow version: 2.17.2
pynrrd version: 1.1.1
clearml version: 1.16.5

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies

**Reply by KumoLiu** [reply, marked as answer]
`GradScaler` moved under torch.amp after torch 2.3.0. Could you please try upgrade your PyTorch version or you can simply use `torch.cuda.amp.GradScaler`?
https://github.com/pytorch/pytorch/tree/v2.3.0/torch/amp

Thanks.

**Reply by shengzhang90** [reply]
Thanks a lot for your responsive answer.
